### ISSUE TITLE
Multiple mapping variables leads to overplotting

### ISSUE THREAD

**Issue by cpsievert** [ropensci/plotly#577]
In this example 64 points are plotted, instead of 32. 

```r
p <- plot_ly(mtcars, x = disp, y = wt, symbol = factor(vs), color = factor(am), mode = "markers") 
l <- plotly_build(p)
sum(sapply(l$data, function(x) length(x$x)))
#> [1] 64
```

It'd be awesome if we could extract out the [existing infrastructure for `ggplotly()` which works this way](https://github.com/ropensci/plotly/blob/d136b69a293f0b4873e5bc991bb05a9afeadbece/R/layers2traces.R#L39-L53), and take advantage of it in `plotly_build()`.

**Comment by cpsievert** [comment]
Actually it might be slightly easier/lighter-weight to take advantage of the new [groupby transform](https://github.com/plotly/plotly.js/pull/499) when it gets merged.

**Comment by cpsievert** [comment]
Closed via d2bf2a1eb56d824a091e2cc9afea09a5968dd050.
